docs/writing-templates.md

```diff
@@ -0,0 +1,84 @@
+# Writing templates
+
+Templates are, as implied from their name, files that define the format of the generated data by this tool.
+
+The tool is designed to leverage any template in the compatible format. This file will guide you through writing template files for any occurrence.
+
+There are 2 supported template types: `placeholder` (fast) and `gotext` (flexible). We suggest to use `gotext` unless performances are **critical** to your workflow.
+
+## How does templating works?
+
+At its core, this tool works by loading field definitions from a file and a template. The template is rendered using information from:
+- the field definitions file
+- the (optional) field generation configurations
+
+This operation has some caveats:
+- generating data is a computing and memory intensive task; you can expect to face computing pressure more easily while memory usage depends on the choosen template type and the template itself; performances of this tool is taken in great consideration, for more information see [performances.md](./performances.md);
+- the use case is generating fake but plausible data, a challenge for a random data generator, as it implies:
+  - in observability dataset usually exhibit high value cardinality; generating high cardinality dataset is a goal, for more information see [cardinality.md](./cardinality.md);
+  - on the opposite side, a data generator may produce data with high similarity, which will not exhibit a behaviour similar to real data; while we aim to increase data dissimilarity, is not a goal to have complete real like data generation;
+  - another characteristic of datasets in observability is high field dimensionality; generating high dimensionality dataset is a goal, for more information see [dimensionality.md](./dimensionality.md);
+- data generation may not be a real time operation executed together/just before using the generated data; we aim to provide corpus of generated data that can be replayed as if they were being emitted in real time with the goal of allowing creation of big and complex datasets.
+
+## Folder structure
+
+All templates are contained in `/assets/templates` folder. Within it there should be a folder for each Integration data stream templates are available from.
```

Comment:
oki with this naming and structure standard: let's just clarify that the name of the folder for the integration data stream must be in the format `fmt.Sprintf("%s.%s", packageName, dataStreamName)`

Reply:
nit: you have both "schema A" and "schema B" assets on https://github.com/elastic/elastic-integration-corpus-generator-tool/pull/43

I will propose to use a specific folder for each schema (like you have `schema-b` but not `schema-a`)

let's update the content of this section once we agree on a format